src/components/textarea/textarea.component.tsx

```diff
@@ -278,6 +287,18 @@ export const Textarea = React.forwardRef(
 
     const hintId = useRef(guid());
 
+    const getFormatNumber = (rawValue: number, locale: string) =>
+      new Intl.NumberFormat(locale).format(rawValue);
+    const capturedLocale = characterCountHint?.match(/{([^']+)}/)?.[1];
+    const formattedCharacterLimit = getFormatNumber(
+      characterLimit as number,
+      capturedLocale as string
+    );
+    const formattedCharacterCountHint = characterCountHint?.replace(
+      /\{.*?\}/g,
+      formattedCharacterLimit
+    );
```

Comment:
**suggestion**: Extending on @edleeks87's suggestion, we could also simplify this code to something like:

```tsx
const getInputHintWithCharacterLimit = (rawString: string, countLimit: number) => {
      const capturedLocale = rawString.match(/{([^']+)}/)?.[1];
      const format = new Intl.NumberFormat(capturedLocale).format(countLimit);
      return rawString.replace(/\{.*?\}/g, format);
};
```

which would allow us to do this when rendering the input hint:

```ts
const formattedHint = (inputHint && characterLimit) ? getInputHintWithCharacterLimit(inputHint, characterLimit) : inputHint;
```